tools/expotools/src/commands/GenerateSDKDocs.ts

```diff
@@ -67,6 +67,22 @@ async function action(options) {
     );
 
     await fs.copy(path.join(SDK_DOCS_DIR, 'unversioned'), targetSdkDirectory);
+
+    // Version the sourcecode URLs for the API pages
+    const apiPages = await fs.readdir(path.join(targetSdkDirectory, 'sdk'));
+    apiPages.forEach(async (api) => {
+      const apiFilepath = path.join(targetSdkDirectory, 'sdk', api);
+      let content = await fs.readFile(apiFilepath, 'utf8');
+      const sourceCodeUrl = content.match(/.*sourceCodeUrl.*\n/);
+      if (sourceCodeUrl) {
+        sourceCodeUrl[0] = sourceCodeUrl[0].replace(
+          /\/tree\/[^/]*\//,
+          `/tree/sdk-${sdk.substring(0, 2)}/`
+        );
+        content = content.replace(/.*sourceCodeUrl.*\n/, sourceCodeUrl[0]);
+        await fs.writeFile(apiFilepath, content, 'utf8');
+      }
```

Comment:
I think we should be more strict in regexp matchers, what if I use `sourceCodeUrl` word somewhere in the doc? 😄 Also, I made a function `transformFileAsync` in `src/Utils.ts` that might be handy here. Let me know if something 👇 needs more explanation!

```suggestion
      await transformFileAsync(apiFilePath, [
        {
          pattern: /^(sourceCodeUrl:.*\/tree\/)([^\]+)(\/.*)$/m,
          replaceWith: `$1${sdk.substring(0, 2)}$3`,
        },
      ]);
```

Reply:
> what if I use sourceCodeUrl word somewhere in the doc

since it grabs the first match, that shouldn't be a worry (sourcodeUrl is always the second line in the doc, after title)

> Also, I made a function transformFileAsync in src/Utils.ts that might be handy here

Awesome! will use this

Reply:
> since it grabs the first match, that shouldn't be a worry (sourcodeUrl is always the second line in the doc, after title)

Yeah, it does grab the first match, but it also grabs the longest possible match 😉 Since you used `.*` it will match everything before and as a result it will match the last `sourceCodeUrl` 😄 It would be better to use "lazy mode" here – so instead of `.*` you can use `.*?` which matches the shortest possible substring.